qiskit_machine_learning/algorithms/classifiers/vqc.py

```diff
@@ -132,13 +136,18 @@ def __init__(
         super().__init__(
             neural_network=neural_network,
             loss=loss,
-            one_hot=True,
             optimizer=optimizer,
             warm_start=warm_start,
             initial_point=initial_point,
             callback=callback,
         )
 
+        # For checking the number of classes is the same for different batches.
+        self.warm_start: bool = warm_start
+        self.num_classes: int | None = None
```

Comment:
- I originally had it as a read-only protected member, but then just decided to expose it directly. I'm not sure why; I think because I was working on the factories in Nature which are being refactored to directly expose their attributes. I'll change it back to read-only.
- This is to address #322. The instance-level one is just to compare against prior batches, and for the user to have read-only access for debugging, etc (when I've protected it). It should stay the same for all batches if `warm_start=True`. If `warm_start=False` then it's just overridden by the next fit (as it should represent a whole new trained model anyway in that case, I think).
- Yes, I think so due to the logic in the bullet point above, which is encoded in `_validate_num_classes`.

Reply:
If you're happy with the need for it, I could add some comments to make it clearer?